src/shadow-embed.js

```diff
@@ -314,6 +316,40 @@ function getStylesheetRules(doc, css) {
   }
 }
 
+/**
+ * @param {!ShadowRoot} shadowRoot
+ * @param {string} name
+ * @param {string} cssText
+ */
+export function installShadowStyle(shadowRoot, name, cssText) {
+  const doc = shadowRoot.ownerDocument;
+  const win = toWin(doc.defaultView);
+  if (shadowRoot.adoptedStyleSheets !== undefined &&
+      CSSStyleSheet.prototype.replaceSync !== undefined) {
+    const cache = win[SHADOW_CSS_CACHE] || (win[SHADOW_CSS_CACHE] = {});
+    let styleSheet = cache[name];
```

Comment:
potentially silly question: is there any harm in using the `cssKey` as a cache key?

Reply:
What's `cssKey`?

Reply:
Sorry, typo, meant `cssText`

Reply:
Yeah. We could do that. Would you prefer? Overall I think that'd have benefits. A longer `cssText` might need more CPU to compute hash. OTOH, it's somewhat unlikely that there'd be overlap between different extensions.

Reply:
The only benefit for `cssText` is so we don't need to pass in `name`, but from a debugging standpoint, and even a clarity standpoint, it makes more sense to keep name.

Reply:
Yeah. This is about where I ended up as well.